.github/workflows/build_windows_exe.yml

```diff
@@ -0,0 +1,44 @@
+name: build-windows-exe
+
+on:
+  pull_request:
+
+  # TODO: Add a release event to trigger the workflow
+  # release:
+  #   types: [ published ]
+
+jobs:
+
+  deploy:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [ '3.10' ]
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Build package
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements-dev.txt
+        python -m build
+        pyinstaller dicom-validator.spec -y
+
+    - name: Archive artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: windows_exe
+        path: dist/*.exe
+    # TODO: How to Publish?
+    # - name: Publish package to PyPI
+    #   uses: pypa/gh-action-pypi-publish@release/v1
+    #   with:
+    #     user: __token__
+    #     password: ${{ secrets.PYPI_TOKEN_VALIDATOR }}
```

Comment:
Publish to pypi is already done in the release-deploy workflow - you should probably add this to that workflow.
That workflow is triggered by making a GitHub release.

Reply:
You may need to download the executable artifact into the `/dist` folder before publishing though, as publishing runs under Ubuntu.

Reply:
Since PyPI is meant for distributable python packages, how about publishing the artifacts as part of GitHub release? Right now, I see that the source code is available as zip and tar.gz. How about I add another step to the current `release-deploy.yml` to publish the executables as artifacts using `gh release create`?

Reply:
> how about publishing the artifacts as part of GitHub release?

Actually that was my first thought, I hadn't even thought about publishing it on PyPi. So yes, I agree - that makes certainly sense.

Reply:
Please see PR #110 . If that looks good, we don't need this PR.